README_INTERFACE_TESTS.md

```diff
@@ -0,0 +1,183 @@
+# Interface tests
+
+Charm relation interfaces are information exchange protocols for Juju charms.
+In this repository, under each interface specification, it is possible to store a list of interface tests. The interface maintainers will write and maintain those tests for the interfaces they own. 
+The goal of the interface tests is to determine whether a given charm satisfies an interface protocol.
+
+There are two personas that need to work with interface tests: interface authors and interface users.
+If you are an interface user, i.e. the owner of a charm that uses some of the interfaces specified in this repository, then you should look at the documentation in the [interface-tester-pytest](https://github.com/canonical/interface-tester-pytest) repository.
+
+If you are an interface author, this page is for you.
+
+# How-To write interface tests as an interface author
+
+In order to add interface tests to a charm relation interface `"my-interface"`, you should commit to this repository one or more python files under `repo-root/interfaces/my-interface/v0/interface_tests/`; for example:
+`repo-root/interfaces/my-interface/v0/interface_tests/requirer_tests.py` and `repo-root/interfaces/my-interface/v0/interface_tests/provider_tests.py`.
+
+Any python file in that directory will be scraped for tests. A test is any function decorated with the `interface_tester.interface_test_case` decorator.
+Before you start writing interface tests, it's a good idea to pip install [interface-tester-pytest](https://github.com/canonical/interface-tester-pytest).
+That will give you the right type hints as you're typing the code, and will also give you a little utility cli to verify that the tests you are writing are being correctly picked up by the same code that will eventually be used to run them.
+
+Suppose you want to contribute interface tests for the `ingress` interface.
+Some tests are meant for the requirer, some for the provider. 
+
+Simplifying a bit, the `ingress` protocol implemented by e.g. Traefik contains some norms for the requirer, and some for the provider:
+- on `ingress-relation-joined`, the requirer should write its part of the data to the relation. 
+- on `ingress-relation-changed`, if the provider has written its part of the data to the relation, then the provider should publish its part of the data; otherwise it should publish no data.
+
+
+## A minimal test case
+
+An interface test case is the direct translation of one such statement into a scenario test. For example:
```

Comment:
Avoid explanations in a how-to. The goal of a how-to is not to learn stuff but to help in development.